web-dev/src/components/HomePage/HomePageComponent.js

```diff
@@ -1,55 +1,34 @@
-import { React, useState, useEffect, useCallback } from "react";
+import './home-page.css';
 import HomeBannerComponent from "../HomeBanner/HomeBannerComponent.js";
 import NavigationTopBar from "../NavigationTopBar/index.js";
 import BoxArtList from "../BoxArtList/index.js"
-import * as bookmarksService from '../../services/bookmarks-service.js';
+import { React, useState, useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
 import { useAuth } from "../../contexts/auth-context";
-import GetMovies from "./GetMovies.js"
-import './home-page.css';
-
+import { findAllBookmarkedByUser } from "../../actions/bookmark-actions.js";
+import * as moviesService from '../../services/movies-service.js';
 
 
 const HomePageComponent = () => {
     const { getUserData } = useAuth();
-    const [bookmarks, setBookmarks] = useState([]);
+    const currentUser = getUserData();
     const [trending, setTrending] = useState([]);
-    const [currentUser, setUser] = useState('');
+    const dispatch = useDispatch()
+    const numMovies = 6;
+    let isBooked = false;
 
-    // A user must be logged in to see the bookmarks panel
-    useEffect(() => {
-        async function getUserProfile() {
-            try {
-                const currentUser = getUserData();
-                setUser(currentUser);
-            } catch (e) {
-            };
-        }
-        getUserProfile();
-    }, [getUserData]);
+    // set bookmarks from the redux state
+    const bookmarks = useSelector(state => state.bookmarks);
+    // update bookmarks once the state changes
+    useEffect(() => findAllBookmarkedByUser(dispatch, currentUser.userID), [dispatch, currentUser.userID]);
 
     // Retrieve a random sample of movies from the database only once
     useEffect(() => {
-        GetMovies().then(result => setTrending(result));
```

Comment:
I refactored the GetMovies so it's cleaner...it was acting as a wrapper function which was pretty pointless.